ambry-api/src/main/java/com/github/ambry/config/QuotaConfig.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.config;
+
+import com.github.ambry.quota.QuotaMode;
+import com.github.ambry.utils.Utils;
+import java.util.HashSet;
+import java.util.Set;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+
+/**
+ * Config for quota of various resources in Ambry.
+ */
+public class QuotaConfig {
+  public static final String QUOTA_ENFORCER_SOURCE_PAIR_INFO_STR = "quotaEnforcerSourcePairInfo";
+  public static final String ENFORCER_STR = "enforcer";
+  public static final String SOURCE_STR = "source";
+  public static final String QUOTA_CONFIG_PREFIX = "quota.";
+  public static final String REQUEST_QUOTA_THROTTLING_ENABLED = QUOTA_CONFIG_PREFIX + "request.throttling.enabled";
+  public static final String HOST_QUOTA_THROTTLING_ENABLED = QUOTA_CONFIG_PREFIX + "host.throttling.enabled";
+  public static final String QUOTA_THROTTLING_MODE = QUOTA_CONFIG_PREFIX + "throttling.mode";
+  public static final String REQUEST_QUOTA_ENFORCER_SOURCE_PAIR_INFO_JSON =
+      QUOTA_CONFIG_PREFIX + "request.enforcer.source.pair.info.json";
+  public static final String HOST_QUOTA_ENFORCER_FACTORIES = QUOTA_CONFIG_PREFIX + "host.enforcer.factories";
+  public static final String QUOTA_MANAGER_FACTORY_CLASS = QUOTA_CONFIG_PREFIX + "manger.factory.class";
+  public static final String DEFAULT_QUOTA_MANAGER_FACTORY_CLASS = "com.github.ambry.quota.AmbryQuotaManagerFactory";
+  public static final String DEFAULT_QUOTA_THROTTLING_MODE = QuotaMode.TRACKING.name();
+  public StorageQuotaConfig storageQuotaConfig;
+
+  /**
+   * Config to enable throttling on customer's account or container.
+   */
+  @Config(REQUEST_QUOTA_THROTTLING_ENABLED)
+  @Default("false")
+  public boolean requestQuotaThrottlingEnabled;
+
+  /**
+   * Config to enable throttling for host's hardware resources quota.
```

Comment:
@justinlin-linkedin Added a comment to indicate this will enable HostLevelThrottler.